### ISSUE TITLE
Timezones are wrong

### ISSUE THREAD

**Issue by ale5000-git** [rmyorston/busybox-w32#454]
Bash online:
```
$ TZ='Europe/Rome' date
Thu Sep 19 05:18:14 AM CEST 2024
$ TZ='CET' date
Thu Sep 19 05:18:14 AM CEST 2024
```

BusyBox for Windows:
```
$ TZ='Europe/Rome' date
Thu Sep 19 04:18:14 ope 2024
$ TZ='CET' date
Thu Sep 19 03:18:14 CET 2024

```

There are 2 errors: the hour and the timezone string.
Is it possible to fix them?

**Comment by rmyorston** [comment]
Yeah, Microsoft's timezone handling is a bit rubbish. It's almost as if they hadn't read POSIX properly ;-)

What _sort of_ works is specifying the time zone in full, with offset and DST name:
```
~ $ date -u
Thu Sep 19 07:43:02 UTC 2024
~ $ TZ=CET-1CEST date
Thu Sep 19 09:43:04 CES 2024
~ $
```

**Comment by ale5000-git** [comment]
@rmyorston
Is it possible to implement a custom code for some cases?

My idea is: if the TZ match this regexp `[A-Za-z]{3,}[+-][0-9]*` then use custom code and skip the windows API.
Looking at the bash online the letters have only a minimum limit of 3 but the have no maximum limit so long strings are allowed.
If the number is higher of 24 it is changed to 24.
If the number is missing it is changed to 0.
-1 sum 1.
+1 subtract 1.

Examples:
```
$ TZ='AfgdfhTHTHtHhh+1' date
Thu Sep 19 03:53:56 PM AfgdfhTHTHtHhh 2024
$ TZ='CET-1' date
Thu Sep 19 05:53:56 PM CET 2024
$ TZ='CEST-2' date
Thu Sep 19 06:53:56 PM CEST 2024
```
